contracts/libraries/ProxyDeltas.sol

```diff
@@ -89,16 +79,20 @@ library ProxyDeltasImpl {
         int128 thisCycleDeltaAdded,
         int128 nextCycleDeltaAdded
     ) internal {
-        require(cycle != CYCLE_ROOT && cycle != CYCLE_END, "Invalid cycle number");
-        // Item not attached to the list
-        if (self.data[cycle].next == CYCLE_ROOT) {
```

Comment:
Could you elaborate why using `isAttached` is cheaper? Seems like we need to store more now, no?

Reply:
EVM reads and writes the whole slots anyway so slipping in another variable is practically free. We're greatly simplifying logic, which is good not only for humans, but it also lets the compiler shave off a few hundred gas.